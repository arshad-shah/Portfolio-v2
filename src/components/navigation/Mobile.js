import React from "react";
import { styled } from "@mui/material/styles";
import { Box, List, ListItem, Link as ExternalLink } from "@mui/material";
import { Link } from "react-scroll";
import Resume from "../../assets/ArshadResume.pdf";

const PREFIX = "Mobile";

const classes = {
  button: `${PREFIX}-button`,
  mobileMenuList: `${PREFIX}-mobileMenuList`,
  mobileMenuItem: `${PREFIX}-mobileMenuItem`,
};

const StyledBox = styled(Box)(({ theme }) => ({
  [`& .${classes.button}`]: {
    "&:hover": {
      backgroundColor: "transparent",
    },
    color: theme.palette.text.primary,
    fontWeight: "bold",
    textDecoration: "none",
    cursor: "pointer",
    borderRadius: "0.5rem",
    transition: "all 0.5s ease",
    [theme.breakpoints.down("sm")]: {
      fontSize: "1rem",
    },
    [theme.breakpoints.down("smxs")]: {
      fontSize: "0.8rem",
    },
  },

  [`& .${classes.mobileMenuList}`]: {
    display: "flex",
    flexDirection: "row",
    bottom: "0",
    position: "fixed",
    width: "100%",
    backgroundColor: theme.palette.primary.main,
    zIndex: "100",
    padding: "10px",
  },

  [`& .${classes.mobileMenuItem}`]: {
    justifyContent: "center",
    padding: "10px",
  },
}));

export default function Mobile() {
  return (
    <StyledBox aria-label="mobileMenu" data-toggle="mobileMenu">
      <List className={classes.mobileMenuList}>
        <ListItem button className={classes.mobileMenuItem}>
          <Link
            className={classes.button}
            data-block="Aboutmemenuitemmobile"
            to="aboutme"
            data-testid="aboutmeOption"
            spy
            smooth
            aria-label="About Me"
          >
            ABOUT ME
          </Link>
        </ListItem>
        <ListItem button className={classes.mobileMenuItem}>
          <Link
            className={classes.button}
            data-block="Projectsmenuitemmobile"
            to="projects"
            data-testid="projectsOption"
            spy
            smooth
            aria-label="Projects"
          >
            PROJECTS
          </Link>
        </ListItem>
        <ListItem button className={classes.mobileMenuItem}>
          <ExternalLink
            className={classes.button}
            data-testid="resumeOption"
            variant="outlined"
            href={Resume}
            data-block="Resumemenuitemmobile"
            aria-label="Resume"
          >
            RESUME
          </ExternalLink>
        </ListItem>
      </List>
    </StyledBox>
  );
}
