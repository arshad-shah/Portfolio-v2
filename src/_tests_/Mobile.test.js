import React from 'react';
import { render, screen } from '@testing-library/react';
import Mobile from '../components/navigation/Mobile';

describe('Mobile layout of the site:', () => {
	it('should render Mobile menu with the About me link', () => {
		render(<Mobile />);
		const aboutme = screen.getByText('About Me');
		expect(aboutme).toBeInTheDocument();
	});

	it('should render Mobile menu with the Projects link', () => {
		render(<Mobile />);
		const Projects = screen.getByText('Projects');
		expect(Projects).toBeInTheDocument();
	});

	it('should render Mobile menu with the Resume link', () => {
		render(<Mobile />);
		const Resume = screen.getByText('Resume');
		expect(Resume).toBeInTheDocument();
	});

	it('should render Mobile', () => {
		const { container } = render(<Mobile />);
		expect(container).toMatchSnapshot();
	});
});
