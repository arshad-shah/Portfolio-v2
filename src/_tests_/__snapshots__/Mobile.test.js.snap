// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile layout of the site: should render Mobile 1`] = `
<div>
  <div
    aria-label="mobileMenu"
    class="MuiBox-root MuiBox-root-16"
    data-toggle="mobileMenu"
  >
    <ul
      class="MuiList-root makeStyles-mobileMenuList-14 MuiList-padding"
    >
      <li
        class="MuiListItem-root makeStyles-mobileMenuItem-15 MuiListItem-gutters"
      >
        <a
          aria-label="About Me"
          class="makeStyles-button-13"
          data-block="Aboutmemenuitemmobile"
          data-testid="aboutmeOption"
        >
          <span
            class="MuiTypography-root MuiTypography-button"
          >
            About Me
          </span>
        </a>
      </li>
      <li
        class="MuiListItem-root makeStyles-mobileMenuItem-15 MuiListItem-gutters"
      >
        <a
          aria-label="Projects"
          class="makeStyles-button-13"
          data-block="Projectsmenuitemmobile"
          data-testid="projectsOption"
        >
          <span
            class="MuiTypography-root MuiTypography-button"
          >
            Projects
          </span>
        </a>
      </li>
      <li
        class="MuiListItem-root makeStyles-mobileMenuItem-15 MuiListItem-gutters"
      >
        <a
          aria-label="Resume"
          class="makeStyles-button-13"
          data-block="Resumemenuitemmobile"
          data-testid="resumeOption"
          href="ArshadResume.pdf"
          variant="outlined"
        >
          <span
            class="MuiTypography-root MuiTypography-button"
          >
            Resume
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
